salt/states/file.py

```diff
@@ -312,18 +313,36 @@ def _gen_keep_files(name, require):
     Generate the list of files that need to be kept when a dir based function
     like directory or recurse has a clean.
     '''
+    def _is_subdir(path, directory):
+        path = os.path.realpath(path)
+        directory = os.path.realpath(directory)
+
+        relative = os.path.relpath(path, directory)
+
+        return (not relative.startswith(os.pardir))
```

Comment:
I find this method a little bit misleading. From its description it means that if I check "sysconfig" is a subdir of "/etc", it should return me True, while it surely does not. So it is as minimum a wrong name of the function, or a bug.